src/plcc/load_spec/parse_spec/parse_semantic_spec/parse_targetLocator_test.py

```diff
@@ -0,0 +1,25 @@
+from pytest import raises
+from .parse_targetLocator import TargetLocator, parse_targetLocator, InvalidTargetLocatorError
+from plcc.load_spec.load_rough_spec.parse_lines import Line
+
+def test_ignore_EOL_comments():
+    line = Line('Class:init #comment', 1, None)
+    assert parse_targetLocator(line) == TargetLocator(line, 'Class', 'init')
+
+def test_requires_colon():
+    with raises(InvalidTargetLocatorError):
+        targetLocator = parse_targetLocator(Line('Class init', 1, None))
+
+def test_modifier():
+    lines = [Line('Class:init', 1, None),Line('Class:modifier', 1, None)]
+    assert parse_targetLocator(lines[0]) == TargetLocator(lines[0], 'Class', 'init')
+    assert parse_targetLocator(lines[1]) == TargetLocator(lines[1], 'Class', 'modifier')
+
+def test_empty_line_raises_error():
+    with raises(InvalidTargetLocatorError):
+        parse_targetLocator(Line('', 1, None))
+
+def test_className():
+    lines = [Line('Class', 1, None), Line('name', 2, None)]
+    assert parse_targetLocator(lines[0]) == TargetLocator(lines[0], 'Class', None)
+    assert parse_targetLocator(lines[1]) == TargetLocator(lines[1], 'name', None)
```

Comment:
Once again, the objects within the tests should really be defined in a single method, as opposed to having each test define them. This can be done for all objects that you are instantiating in these tests.